pkgs/development/python-modules/arelle/default.nix

```diff
@@ -1,87 +1,144 @@
 {
-  gui ? true,
+  lib,
   buildPythonPackage,
   fetchFromGitHub,
-  lib,
+  pythonOlder,
+
+  setuptools,
+  setuptools-scm,
+
   sphinx,
-  lxml,
+  sphinx-autodoc2,
+  myst-parser,
+  sphinx-copybutton,
+  furo,
+
+  certifi,
+  filelock,
   isodate,
+  lxml,
   numpy,
   openpyxl,
-  tkinter ? null,
-  py3to2,
-  isPy3k,
-  python,
-  ...
+  pyparsing,
+  python-dateutil,
+  regex,
+  tkinter,
+
+  pycryptodome,
+  pg8000,
+  pymysql,
+  pyodbc,
+  rdflib,
+  holidays,
+  pytz,
+  tinycss2,
+  graphviz,
+  cheroot,
+  cherrypy,
+  tornado,
+
+  pytestCheckHook,
+  boto3,
 }:
 
 buildPythonPackage rec {
-  pname = "arelle${lib.optionalString (!gui) "-headless"}";
-  version = "18.3";
-  format = "setuptools";
+  pname = "arelle";
+  version = "2.30.25";
+  pyproject = true;
 
-  disabled = !isPy3k;
+  disabled = pythonOlder "3.8";
 
-  # Releases are published at http://arelle.org/download/ but sadly no
-  # tags are published on github.
   src = fetchFromGitHub {
     owner = "Arelle";
     repo = "Arelle";
-    rev = "edgr${version}";
-    sha256 = "12a94ipdp6xalqyds7rcp6cjwps6fbj3byigzfy403hlqc9n1g33";
+    rev = "refs/tags/${version}";
+    hash = "sha256-xzTrFie97HDIqPZ4nzCh+0p/w0bTK12cS0FSsuIi7tY=";
   };
+
   outputs = [
     "out"
     "doc"
   ];
-  patches = [ ./tests.patch ];
-  postPatch = "rm testParser2.py";
-  nativeBuildInputs = [
+
+  postPatch = ''
+    substituteInPlace pyproject.toml --replace-fail \
+        'requires = ["setuptools~=73.0", "wheel~=0.44", "setuptools_scm[toml]~=8.1"]' \
+        'requires = ["setuptools", "wheel", "setuptools_scm[toml]"]'
+  '';
+
+  build-system = [
+    setuptools
+    setuptools-scm
+
+    # docs
     sphinx
-    py3to2
+    sphinx-autodoc2
+    myst-parser
+    sphinx-copybutton
+    furo
   ];
-  propagatedBuildInputs = [
-    lxml
+
+  dependencies = [
+    certifi
+    filelock
     isodate
+    lxml
     numpy
     openpyxl
-  ] ++ lib.optionals gui [ tkinter ];
-
-  # arelle-gui is useless without gui dependencies, so delete it when !gui.
-  postInstall =
-    lib.optionalString (!gui) ''
-      find $out/bin -name "*arelle-gui*" -delete
```

Comment:
I did make an attempt to fix the packages as part of ZHF, where I changed this:

```diff
-      find $out/bin -name "*arelle-gui*" -delete
+      find $out/bin -name "*arelleGUI*" -delete
```

If that doesn't get the job done (might not get rid of unneeded runtime deps?), then it's fine to just drop the headless variation.